### PR TITLE
Add Vega Typings

### PR DESCRIPTION
**Pull request by domoritz**
Fixes vega/vega-typings#58

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# top-most EditorConfig file
```
Comment (jheer) [inline .editorconfig]: As this adds a new top-level config file, can you say more about what it does and what (if anything, such as a plug-in) is needed for editors to adhere to it?

File: package.json [hunk 2]
```diff
@@ -24,8 +24,8 @@
     "rollup-plugin-json": "^3.1.0",
     "rollup-plugin-node-resolve": "^4.0.0",
     "tape": "^4.9.1",
-    "tv4": "^1.3.0",
     "terser": "^3.11.0",
+    "tv4": "^1.3.0",
```
Comment (jheer) [inline package.json]: This rearrangement makes sense but also caused a merge conflict. These kinds of orthogonal changes to the main intent of the PR are best included as a separate PR altogether.

File: packages/vega-typings/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "repository": "vega/vega",
+  "author": "Dominik Moritz <domoritz@gmail.com>",
+  "keywords": [
+    "vega",
+    "typescript",
+    "typings"
+  ],
+  "license": "BSD-3-Clause",
+  "devDependencies": {
+    "dtslint": "^0.4.2",
```
Comment (jheer) [inline packages/vega-typings/package.json]: Although only used by vega-typings at the moment, I'm fine having these devDependencies moved to the top-level package.json file for consistency across packages and easier updates to the development environment.

**Comment by jheer** [comment]
I merged the earlier dependency update PR, but now there is a conflict here. @domoritz, can you update to resolve the conflict first?

File: .editorconfig (hunk 1 above)
Comment (domoritz) [reply .editorconfig]: This sets consistent whitespace characters and line endings for various editors. We use it in Vega-Lite but I'm happy to remove it from this PR. 

Some editors (e.g. github, vs) don't need any additional plugins. There are plugins for various editors at https://editorconfig.org/#download.

File: package.json (hunk 2 above)
Comment (domoritz) [reply package.json]: This might have been yarn. I'll rebase.

File: .editorconfig (hunk 1 above)
Comment (jheer) [reply .editorconfig]: I'm all for having a useful config and this looks totally reasonable. I just wanted to understand it before merging. I'm also open to using prettier in a consistent way across the Vega modules (though obviously that would be the subject of a different PR).

**Comment by domoritz** [comment]
Rebased and tests are passing.

**Pull request merged**